fast/stages/1-resman/tenant-logging.tf

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+# tfdoc:file:description Audit log project and sink for tenant root folder.
+
+locals {
+  log_sink_destinations = merge(
+    # use the same dataset for all sinks with `bigquery` as  destination
+    {
+      for k, v in local.log_sinks :
+      k => module.log-export-dataset[0] if v.type == "bigquery"
+    },
+    # use the same gcs bucket for all sinks with `storage` as destination
+    {
+      for k, v in local.log_sinks :
+      k => module.log-export-gcs[0] if v.type == "storage"
+    },
+    # use separate pubsub topics and logging buckets for sinks with
+    # destination `pubsub` and `logging`
+    module.log-export-pubsub,
+    module.log-export-logbucket
+  )
+  log_sinks = (
+    length(var.logging.log_sinks) > 0 || var.root_node == null
+    ? var.logging.log_sinks
+    # provide default log sinks to tenants
+    : {
+      audit-logs = {
```

Comment:
We're capturing audit logs here and in the tenant core folders. Do we really want to have it in both places?

Related: [intercepting sinks](https://cloud.google.com/logging/docs/export/aggregated_sinks#:~:text=An%20intercepting%20aggregated%20sink%20prevents%20logs%20from%20being%20routed%20through%20sinks%20in%20child%20resources%2C)

Reply:
the core folder sink is a single one, just for audit logs, and it's controlled by the central team and logging to the central IaC project: it gives the central team a way to easily check/debug tenants
the sinks here are created in the tenant folder and should be defined by the tenant, they are not controlled by the central team